### ISSUE TITLE
[BUG]kWaveSimulation.py

### ISSUE THREAD

**Issue by Liu2021xx** [waltsims/k-wave-python#632]
Due to this line of code self.sensor = kSensor(mask=torch.ones((Nx, Ny, max(1, Nz))), record=["p_final"]) ，Calling a property that does not belong to this class later resulted in an error，such as the self. sensor.time.rversal-boundary_data property 
def time_rev(self):
        """
        Returns:
            True for time reversal simulaions using sensor.time_reversal_boundary_data

        """
        if self.sensor is not None and not isinstance(self.sensor, NotATransducer):
            if not self.options.simulation_type.is_elastic_simulation() and **self.sensor.time_reversal_boundary_data** is not None:
                return True
        else:
            return self.userarg_time_rev

**Comment by waltsims** [comment]
Closing as non-reproducible on current master. `kSensor.time_reversal_boundary_data` is defined at `kwave/ksensor.py:75-81` and returns `None` when unset — accessing it on a numpy-masked `kSensor(mask=..., record=['p_final'])` works fine end-to-end.

I haven't been able to reproduce the AttributeError you described. The original snippet uses `torch.ones(...)` as the mask, which may trigger different attribute-resolution behaviour somewhere downstream, but I can't isolate the failing path without your full env.

If you're still hitting this on current master (post-#705 / #712), please reopen with a runnable minimal reproducer that produces the traceback. A small standalone script using only numpy (or showing the exact torch version + the full traceback) would let us trace it precisely. Thanks for filing the original report!
